protocol/index.ts

```diff
@@ -729,6 +777,26 @@ export interface ArcjetEmailRule<Props extends { email: string }>
   allowDomainLiteral: boolean;
 }
 
+export type CustomDetect<Custom extends string> = (
+  tokens: string[],
+) => (SensitiveInfoEntity<Custom> | undefined)[];
+
+export type SensitiveInfoConfig<Custom extends string> = {
+  allow?: SensitiveInfoEntity<Custom>[];
+  deny?: SensitiveInfoEntity<Custom>[];
+  customDetect?: CustomDetect<Custom>;
+  contextWindowSize: number;
+};
```

Comment:
These aren't user-facing types so they shouldn't exist in the protocol. There's also no real reason to be creating new types for analyze when the generated types work pretty well.